source/auth/auth.rst

```diff
@@ -683,6 +683,176 @@ mechanism
 mechanism_properties
 	MUST NOT be specified.
 
+MONGODB-IAM
+~~~~~~~~~~~
+
+:since: 4.4
+
+MONGODB-IAM authenticates using AWS IAM credentials or temporary IAM credentials 
+assigned to an EC2 machine or ECS task.
+
+MONGODB-IAM requires that a client create a randomly generated nonce. It is 
+imperative, for security sake, that this be as secure and truly random as possible. 
+
+All messages between MongoDB clients and servers are sent as base64 encoded BSON V1.1 Objects. 
+All fields in these messages have a "short name" which is used in the serialized 
+BSON representation and a human readable "friendly name" which is used in this document. They are as follows:
+
+==== ==================== ========== ============================================================================================================================================== 
+Name Friendly Name        Type       Description
+==== ==================== ========== ==============================================================================================================================================
+r    client nonce         BinData    32 byte cryptographically secure random number 
+p    gs2-cb-flag          32-bit int The integer representation of the ASCII charater 'n' or 'y', i.e., ``110`` or ``121``
```

Comment:
If this is a standard BSON-spec compliant 32-bit integer, I would recommend using the same notation as http://bsonspec.org/spec.html
`32-bit int` --> `int32`

Reply:
Done

Reply:
@prashantmital please comment back or resolve.